src/main/java/com/google/sps/servlets/UserFoldersServlet.java

```diff
@@ -19,67 +20,80 @@
 import javax.servlet.http.HttpServletResponse;
 
 import com.google.sps.data.Folder;
-import java.util.List; 
+import java.util.List;
 import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
 
-
-/** **/
 @WebServlet("/userfolders")
 public class UserFoldersServlet extends HttpServlet {
 
-  /**
-  * Query all folders from current user
-  */
+  private static final String CREATE_FORM_KEY = "showCreateFormStatus";
+  private static final String USER_FOLDER_KEY = "userFolders";
+
+  /*
+   * Query all folders from current user
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
     UserService userService = UserServiceFactory.getUserService();
 
-    if (userService.isUserLoggedIn()) {
+    Map<String, Object> jsonInfo = new HashMap<>();
+    List<Folder> userFolders = new ArrayList<>();
+    jsonInfo.put(CREATE_FORM_KEY, false);
 
-      List<Folder> userFolders = new ArrayList<>();
-
-      String userKey = request.getParameter("userKey");
+    if (userService.isUserLoggedIn()) {
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      String userKey = getUserKey(userService, datastore);
 
-      // Query all folders identified by the userKey
       Query folderQuery = new Query("Folder").setAncestor(KeyFactory.stringToKey(userKey));
-
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       PreparedQuery results = datastore.prepare(folderQuery);
 
-      // Add folders to a list
       if (results != null) {
-        for (Entity entity: results.asIterable()) {
-          Folder folder = Folder.EntityToFolder(entity);
+        for (Entity entity : results.asIterable()) {
+          Folder folder = new Folder(entity);
+          folder.setFolderKey(KeyFactory.keyToString(entity.getKey()));
```

Comment:
Why is this line here? Why can't this be done in the constructor?

Reply:
I restructured this in tdd5